src/routes/builder/index.svelte

```diff
@@ -145,358 +15,181 @@
-			value: field,
-			disabled:
-				!config.dataset
-				|| [typeDicts, datasetDicts, aggDicts]
-				.some(isMissing('fields', field))
-		}));
-
-		let activeAxes = computeRequestBody(config);
-
-		if (typeOptions.some(i => i.effaced
-			&& i.value === selectedAxisConfig.type)) {
-			cleanRequestBody();
-		}
-
-		axisOptions.forEach((o,i) => {
-			o.disabled = i >= activeAxes
-		});
-		axisOptions = axisOptions;
-		responsePromise = Promise.resolve(undefined);
-
-		if (IS_BROWSER && window.ts) {
-			if (!datasetTypings) {
-				datasetTypings = await request(
-					'GET',
-					'dsl/datasets.ts',
-					{type:'text'}
-				);
-			}
-			if (selectedParams.output) {
-				const ds = DATASETS[config.dataset].id;
-				const code = `
-					const selection: Aggs<${ds}, '${selectedAxisConfig.field}'> = 
-						${JSON.stringify(selectedParams.pureOutput)};
-				`;
-				const fullCode = datasetTypings + code;
-				selectedFieldCompletions = getCompletions(
-					fullCode,
-					fullCode.lastIndexOf('{') + 1
-				).sort((a, b) => b.required - a.required);
-			}
+	function setAggDocs (agg) {
+		if (aggCompletions && agg in aggCompletions) {
+			routeMachine.send('AGG_DOC_SHOWN', {
+				docstring: aggCompletions[agg]
+			});
+		} else {
+			routeMachine.send('AGG_DOC_DEFAULT');
 		}
 	}
 
 	function getFieldValue (name) {
-		return selectedParams.input[name]
-			|| selectedParams
-			.output[selectedAxis][selectedAxisConfig.aggregation][name];
-	}
-
-	function updateField (name, newValue) {
-		// TODO For text types, should we distinguish between
-		// empty strings and `null` or `undefined`?
-		if (newValue !== null) {
-			selectedParams.input[name] = newValue;
+		if ($params) {
+			return $params[name];
+			// || $selectedParams
+			// .output[$selectedForm.value][$activeSelection.aggregation][name];
 		}
-		else {
-			delete selectedParams.input[name];
-		}
-		computeRequestBody(queryConfig);
-	}
-
-	const cache = {};
-	function doQuery (query) {
-		if (readyForRequest) {
-			const endpoint = getEndpointURL(DATASETS[queryConfig.dataset]);
-			const url = `${endpoint}/_search`;
-			const cacheKey = `${url}/${JSON.stringify(query)}`;
-			if (cacheKey in cache) {
-				responsePromise = Promise.resolve(cache[cacheKey]);
-			} else {
-				responsePromise = request('POST', url, {data: query});
-				responsePromise.then(json => {
-					cache[cacheKey] = json;
-				});
-			}
-		}
-	}
-
-	function axisChanged (newAxis) {
-		selectedAxisConfig = queryConfig.axes[newAxis];
-		selectedParams = axisParams[newAxis];
+		return null;
 	}
 
-	$: axisChanged(selectedAxis);
-	$: !queryConfig.dataset && (selectedAxisConfig.field = null);
-	$: computeLists(queryConfig);
-	$: $selectedRequestTab === 'fields'
-		&& runQueryOnSelect
-		&& doQuery(queryTemplate);
-	$: $selectedRequestTab === 'request'
-		&& runQueryOnSelect
-		&& doQuery(parsedQuery);
+	onMount(() => {
+		routeMachine.send("READY");
+		routeMachine.send("FIELD_DOC_DEFAULT");
+		routeMachine.send("AGG_DOC_DEFAULT");
```

Comment:
I'd rather not initialize stores via events on mount but rather use `DEFAULT_FIELD_DOCS` and `DEFAULT_AGG_DOCS` as initial values for the stores `activeDocs` and `aggDocText`, declaring `DEFAULT_FIELD_DOCS` and `DEFAULT_AGG_DOCS` in `_context.js`